### ISSUE TITLE
Flag on spot is in the air

### ISSUE THREAD

**Issue by Fisico** [teeworlds/teeworlds#1732]
Flag on spot is in the air.
![screenshot_2018-11-26_21-11-06](https://user-images.githubusercontent.com/329471/49039794-1817df80-f1c1-11e8-9448-c19940425952.png)

Dropped flag is on the ground.

**Comment by Fisico** [comment]
i think flagspots should be marked on the map or with some graphics and not by floating in the air.

landil want something similar for weapons...
https://github.com/teeworlds/teeworlds/issues/1134

**Comment by Zatline** [comment]
Not so sure about this whole marking everyting but fundametaly it would be the right thing, 
flag spots at very least! ctf5 already does that btw..

**Comment by Dune-jr** [comment]
@Sonix- shouldn't the flag be planted into the ground?

**Comment by Sonix-** [comment]
If dropped yes, meaning its not on his original position and actually “dropped” by someone

**Comment by Fisico** [comment]
i don't think you will notice this litte air gap when you are playing competitive, also on some maps there is grass or bushes that cover it.

whats wrong about marking the flagspot? 
i have to agree markings for every weapon seems a bit overkill, but the flagspot could be marked it is the most important thing in ctf. It would also be good for new players that don't know maps that well. In CS or TF2 the bomb / flag spots are also marked.

A small flagholder or pad in teestyle would be great.

**Comment by oy** [comment]
Adding someone flag spot indicator might be a good idea.

**Comment by Sonix-** [comment]
I'll quote what Magnet said in #1134 
> I agree for no image, but rather make spots obvious by map convention. Please don't dumb the game down.

**Comment by Fisico** [comment]
the unhookable solution is for example not possible on ctf2 and is it a good idea to have the flagspot always unhookable?

**Comment by Dune-jr** [comment]
> the unhookable solution is for example not possible on ctf2 and is it a good idea to have the flagspot always unhookable?

Definitely not, if those decoration tiles are used, there should be *under* the flag, where it does not affect gameplay.

**Comment by Fisico** [comment]
quick idea.
![image](https://user-images.githubusercontent.com/329471/49244954-5277bb80-f411-11e8-8390-0fc870321212.png)

maybe zatline comes up with a more teeish way

**Comment by Fisico** [comment]
more refined version
![image](https://user-images.githubusercontent.com/329471/49247021-8d302280-f416-11e8-8d69-89026fe4030c.png)

**Comment by Landil** [comment]
Has been discussed ever since ... 

And if you find a way the flag holder looks great, even on every possible point in the map ... e.g. on edges ... I will have a look at it and judge it fairly. 

Keep in mind the original flag design is not as easy at it appears on first sight: the flag's asymmetry combined with a symmetrical flag holder, which is centered to a certain tile on the game layer ... check it out ... I tested several designs years ago, none of it looked great, I dropped it. 

Feel free to present some working ingame designs. Then the community will discuss it.

**Comment by Fisico** [comment]
Does not looks so bad i think. 
Will try some more stuff tomorrow.
![screenshot_2018-11-29_21-59-17](https://user-images.githubusercontent.com/329471/49251624-1d279980-f422-11e8-8b41-b6d8f05103c1.png)

**Comment by Zatline** [comment]
An a-symmetric flag by logic would requiere an a-symmetric holder
![932b2e441f35e30c9871cf25940a35bd](https://user-images.githubusercontent.com/344608/49253601-50b8f280-f427-11e8-8993-b30e80ddc124.png)

**Comment by oy** [comment]
Don't think that adding such a flag holder would dumb the game down. The flags are the key gameplay element of ctf. And currently when your own flag is taken there's no indicator where your own flag spot is. Which is bad for beginners or people who don't know the map. Seen people ingame asking what to do/where to go with the flag.
Something should be done to improve that and to make the life easier for beginners. And adding team coloured unhookable tiles near the flag spot is not a good solution. The tiles are usual ones and can be used anywhere. And you have to rely that map creators stick to it and prevent to add fake spots.

**Comment by Dune-jr** [comment]
I agree, however I am not sure if a hardcoded thing is the way to go. Map creators could find ways to indicate the flag spawn in such a way that it fits the theme of the map. That's how it is in some other games.

**Comment by oy** [comment]
Don't think that a flag holder like within the previous posts would interfere with that.
Could also add some kinda indicator to the score hud? So it points out that you're within your flag spot.

**Comment by LordSk** [comment]
I completely agree with @Dune-jr, we could modify the base ctf map pool to add a decorative flag spot.

**Comment by Fisico** [comment]
problem i see.
if you add the flagspot as tile layer in the maps... you have to put it on the front layer, so the falgholder is in front of the flag itself... but than the tee is behind the flagholder and that looks strange... 

i think it would be ok to add them clientside.

**Comment by Zatline** [comment]
>I am not sure if a hardcoded thing is the way to go. Map creators could find ways to indicate the flag spawn in such a way that it fits the theme of the map.

It should definetly be part of game.png and hardcoded think that would be the most consistent way of doing it since we wouln't have to relay on map creator hinting flag spots with tilesets which u know not everyone will do, or even think about in first place.
If done subtle and right it won't break the theme of the map either, cuz if that was true u could say weapons would break the theme just like that.

**Comment by Fisico** [comment]
new idea...
took some inspiration from zatline and added teamcolors.

![image](https://user-images.githubusercontent.com/329471/49309790-ac918300-f4dc-11e8-9675-e8675ff62059.png)

**Comment by Landil** [comment]
if you cannot distinguish the flag holder colors on 800x600 resolution you can drop the details.

**Comment by nheir** [comment]
It can be useful to disable (for map creators) the flagspot mark, if the flagspot is flying on purpose for instance (leaving the creator choosing whatever background tiles to highlight the flagspot)

**Comment by Fisico** [comment]
@Landil can't test it right now... teeworlds does not change resolution... dunno why

and some more variations...
![image](https://user-images.githubusercontent.com/329471/49321161-15412580-f506-11e8-8035-152bbfb68800.png)

**Comment by Dune-jr** [comment]
If we go with that, I would like to see a wooden version, I think it would match the wooden flag better.

**Comment by Fisico** [comment]
@Dune-jr does not looks good at all

![image](https://user-images.githubusercontent.com/329471/49327474-91695680-f562-11e8-8dd8-8228c679bdf5.png)

**Comment by oy** [comment]
@nheir 
But having an invisible/"floating in the air" flag spot is a good example why a automatic flag spot indicator is a good thing.

A flaw with the flag holder is that you can't see right away if the flag is dropped there or at the stand. Before you had the little gap at the bottom.

**Comment by Fisico** [comment]
i dont think it is possible to drop the flag that accurate, right on the spot.

**Comment by nheir** [comment]
@oy 
> But having an invisible/"floating in the air" flag spot is a good example why a automatic flag spot indicator is a good thing.

It's up to the map creator to make the map playable. The idea is only that a flying flag support is weirder than a background tile without mark. A flying flag spot may by also weird but forcing a support is not a solution. If the map is poorly designed, people will not play on it.

**Comment by Dune-jr** [comment]
Agree, forcing this makes any flag placed in a non-standard place ugly-looking. I don't have any example on the top of my head, but there are ice-transparent maps and ctf_space, maybe?

**Comment by oy** [comment]
Don't think that people won't play a map because the flag spots are designed badly when the reast is great. People will play there, when they get to know where the flag spots are. But it's really a hassle for beginners or people who don't know the map.
And I don't think there's a problem with having a flying flag holder when there's already a flying flag. ;)
@Zatline did a bottom part for the flag holder so i wouldn't be cut off.

**Comment by Fisico** [comment]
@Zatline the flagholder should always have the team color, so you see where you have to return the flag.

**Comment by Zatline** [comment]
> @Zatline the flagholder should always have the team color, so you see where you have to return the flag.

no no no I think u misunderstand ..
I mean holder shows team color when flag isn't there but the moment the flag returns and stands on spot the color of the holder goes off, see it like a light that turns on when flag isn't there, wie ne Lichtschranke.

**Comment by Fisico** [comment]
@Zatline yea great idea. But we have to make the colored ring more led like than...

**Comment by Zatline** [comment]
@Fisico not necessarily I don't think it should be a light I was just pointing out that it would behave in a similar way, it is better to imagine .. for simplicity it should not emit any luminance, no gradient, just plain flag color.

**Comment by android272** [comment]
> ![image](https://user-images.githubusercontent.com/329471/49321161-15412580-f506-11e8-8035-152bbfb68800.png)

I like the top left stand way better than the asymmetric stands. It looks better and it does not make any sense to make it asymmetric if we can just offset it in the game.png

> ![906987e59222800e272cb53da303ffec](https://user-images.githubusercontent.com/344608/49323541-192c7200-f51d-11e8-9e84-a0e1a4c14d6a.png)

**Comment by fokkonaut** [comment]
if this is really wanted its up to the mapper to mark a flag spot using the editor but not by hardcoding such shit in the game.

**Comment by nheir** [comment]
The idea is not to add another map object.
It should be enable or disable in the map setting though.

@fokkonaut Can you be more respectful to people trying to improve the game whatever your opinion is on the feature ?

**Comment by Fisico** [comment]
As this seems to get more attention again and I thought about it again…
Maybe add a server side option for mods to tell the client to show the flag holder or not.
For CTF enable it other mods can server side disable the flag holder in their code. Maybe this is a good compromise.

**Comment by Stiopa866** [comment]
This doesn't seem to have gone anywhere...

But if I were to chip in, I think that it is a great idea to have some indicator for the flag stand, but map based probably...

the graphics don't particularly convince me anyway, but that might just be human resistance to change kicking in :P

**Comment by fokkonaut** [comment]
I dislike this idea too, at least if its not done mapping wise

**Comment by Fisico** [comment]
Every single other game has the flagspot marked in some way. Why should it be a problem in Teeworlds?
We could also provide flagspot doodads to mark the spot in maps. But this needs a new layer to be in front of the flag but behind the tee.

**Comment by jxsl13** [comment]
Which other games?

**Comment by fokkonaut** [comment]
> Every single other game has the flagspot marked in some way. Why should it be a problem in Teeworlds?
> We could also provide flagspot doodads to mark the spot in maps. But this needs a new layer to be in front of the flag but behind the tee.
> 
> edit: I still like the idea of implementing the flagspot in the game.png. If people don't like it they can make their own design or remove the flagspots from the game.png completely. Win for everybody.
> 

Not a win for mod makers or flags in the air. This is one of those things that forces a modded server to assume specific things, which should be added inside of the map. Also, that will look way better for other designs if the map maker can decide how to design it.

**Comment by oy** [comment]
@Fisico wrote:
> edit: I still like the idea of implementing the flagspot in the game.png. If people don't like it they can make their own design or remove the flagspots from the game.png completely. Win for everybody.

Agree. Could add a client option to disable it.

**Comment by Stiopa866** [comment]
Client and/or map toggle seems like a good way to go about it?

Map toggle for the mappers and mod makers and client toggle for those who like the current setup more.
In each case, new vanilla players are exposed to where the flags are.

**Comment by AssassinTee** [comment]
Can we check a version where the flag-holder is colored fully in the team-color? I like the slim one from @Zatline the most

**Comment by oy** [comment]
@Stiopa866 sounds good 👍

**Comment by ChillerDragon** [comment]
I do not like the idea in general. It should be kept simple. There are only a hand full of official maps that are being played. They are mostly mirrored and you spawn next to your flag. This way you know also where the enemy flag is. I do not recall ever being confused about where the flag spot is even when playing custom maps for the first time. Also unlike CS where you have to place a bomb on the bomb spot. In teeworlds you do not really have to interact with the flag spot when it is empty. Even if your gameplay tactic is sitting on the flag spot and doing nothing until it returns its not like this has to be done pixel perfect. You can also stand close to the flag without knowing where exactly it is and then pick it up when it spawns. Usually CTF maps are shaped in a way around the flag that makes it quite obvious where it should be.

But if we were to accept the idea could we please do more iterations on the design. To me all of those new flag stands seem like a visual downgrade.

Also it seems limiting. The floating flag was more flexible and allowed mappers to do more things. Like this flag on the wall on ctf_take. That would look odd with a flying flag stand.

![ctf_take_floating_flag](https://github.com/user-attachments/assets/bb61de5d-a49a-4fa1-813b-7f3b85f71836)

**Comment by tempral** [comment]
why not a particles effect that match the flag color, a little light placeholder

**Comment by Dune-jr** [comment]
> why not a particles effect that match the flag color, a little light placeholder

Effects (rather than static tiles) are a bit harder to make work, but very welcome if they're teeish imo
